app/Services/PolygonService.php

```diff
@@ -212,24 +215,25 @@ protected function validateSitePolygonProperties(string $polygonUuid, array $pro
      *
      * @return string UUID of resulting PolygonGeometry
      */
-    protected function transformAndStorePoints($geojson, $sitePolygonProperties): string
+    protected function transformAndStorePoints($geojson, $sitePolygonProperties): array
     {
-        $pointUuids = [];
-        foreach ($geojson['features'] as $feature) {
-            $pointUuids[] = $this->insertSinglePoint($feature);
+        foreach ($geojson['features'] as &$feature) {
+            $currentPointUUID = $this->insertSinglePoint($feature);
+            $feature['properties']['point_id'] = $currentPointUUID;
```

Comment:
Is this assignment working? I don't understand why, but I've had trouble with `&$` in loops and had to resort to using `data_set` instead. But if you've tested and this is working as expected, I'm happy 👍🏻

Reply:
Yes, it is working fine in local. Hope it works everywhere.